Convert some advanced authoring views to Angular

geoffreykwan opened the pull request:
Make sure the advanced authoring views for these component types still work properly
- Animation
- Audio Oscillator
- Concept Map
- HTML
- Outside URL

Closes #334

geoffreykwan committed:
Convert some advanced authoring views to Angular

- Created the EditCommonAdvancedComponent that contains the common
  advanced authoring components that a lot of component types share
- Converted the advanced authoring views for these component types
  - Animation
  - Audio Oscillator
  - Concept Map
  - HTML
  - Outside URL

#334

hirokiterashima commented on src/app/authoring-tool/edit-advanced-component/edit-advanced-component.component.ts:
```diff
@@ -9,12 +11,39 @@ export abstract class EditAdvancedComponentComponent {
     );
   }
+
+  setShowSubmitButtonValue(show: boolean): void {
+    if (show == null || show == false) {
+      this.authoringComponentContent.showSaveButton = false;
+      this.authoringComponentContent.showSubmitButton = false;
+    } else {
+      this.authoringComponentContent.showSaveButton = true;
+      this.authoringComponentContent.showSubmitButton = true;
+    }
```
```suggestion
  setShowSubmitButtonValue(show: boolean = false): void {
    this.authoringComponentContent.showSaveButton = show;
    this.authoringComponentContent.showSubmitButton = show;
```

hirokiterashima reviewed (changes requested):
Functionality works. I have some code improvement suggestions:
1. Now that these components do not use AngularJS translations, we can remove ```'pascalprecht.translate'``` angularjs dependency and config block: 
```
 .config([
    '$translatePartialLoaderProvider',
    ($translatePartialLoaderProvider) => {
      $translatePartialLoaderProvider.addPart('components/animation/i18n');
    }
  ]);
```
from the [animation/audioOscillator/ConceptMap/Html/OutsideUrl]AuthoringComponentModule files.

2. See inline code comment

geoffreykwan committed:
Clean up some authoring advanced view code

- Clean up setShowSubmitButtonValue()
- Remove translation settings in component module files
- Add some styling for spacing

#334
